File: src/main/java/cat/xlagunas/andrtc/push/PushNotificationServiceImpl.java
package cat.xlagunas.andrtc.push;

import java.util.List;

public class PushNotificationServiceImpl implements PushNotificationService {

    final PushNotificationRepository pushNotificationRepository;

    public PushNotificationServiceImpl(PushNotificationRepository pushNotificationRepository) {
        this.pushNotificationRepository = pushNotificationRepository;
    }

    @Override
    public void sendPush(List<String> pushTokenList, PushMessageData message) {
        pushNotificationRepository.sendPush(new PushMessage.Builder()
                .content(message)
                .tokenList(pushTokenList)
                .build());
    }
}

File: src/main/java/cat/xlagunas/andrtc/push/PushNotificationServiceImpl.kt
package cat.xlagunas.andrtc.push

class PushNotificationServiceImpl(internal val pushNotificationRepository: PushNotificationRepository) : PushNotificationService {

    override fun sendPush(pushTokenList: List<String>, message: PushMessageData) {
        pushNotificationRepository.sendPush(PushMessage(pushTokenList, message))
    }
}
